oval-schemas/linux-definitions-schema.xsd

```diff
@@ -2718,6 +2718,150 @@
+               <sch:let name="state_namespace" value="namespace-uri($reffed_state)"/>
+               <sch:assert test="(($state_namespace='http://oval.mitre.org/XMLSchema/oval-definitions-5#linux') and ($state_name='auditdline_state'))">State referenced in filter for <sch:value-of select="name($parent_object)"/> '<sch:value-of select="$parent_object_id"/>' is of the wrong type. </sch:assert>
+             </sch:rule>
+           </sch:pattern>
+         </xsd:appinfo>
+       </xsd:annotation>
+       <xsd:complexType>
+         <xsd:complexContent>
+           <xsd:extension base="oval-def:ObjectType">
+             <xsd:sequence>
+               <xsd:choice>
+                 <xsd:element ref="oval-def:set"/>
+                 <xsd:sequence>
+					<xsd:element name="filepath" type="oval-def:EntityObjectStringType" nillable="true">
+                     <xsd:annotation>
+                       <xsd:documentation>The filepath element specifies the path and name of a file to evaluate.</xsd:documentation>
+					   <xsd:documentation>If the xsi:nil attribute is set to true, then the rules specified are the ones executed at runtime.</xsd:documentation>
+					   <xsd:appinfo>
+					   <sch:pattern id="linux-def_auditfilepath">
+							<sch:rule context="linux-def:auditdline_object/linux-def:filepath">
+								<sch:assert test="not(@operation) or @operation='equals'"><sch:value-of select="../@id"/> - operation attribute for the filepath entity of an auditdline_object should be 'equals', note that this overrules the general operation attribute validation (i.e. follow this one)</sch:assert>
+							</sch:rule>
+						</sch:pattern>
+					   </xsd:appinfo>
+                     </xsd:annotation>
+                   </xsd:element>
+                   <xsd:element name="field" type="oval-def:EntityObjectStringType" minOccurs="0" maxOccurs="1">
+                     <xsd:annotation>
+                       <xsd:documentation>As described in the auditctl(8) manpage, the field is an association of name, operation and value used to parameter the audit rule</xsd:documentation>
+                     </xsd:annotation>
+                   </xsd:element>
+				   <xsd:element name="syscall" type="oval-def:EntityObjectStringType" minOccurs="0" maxOccurs="1">
+                     <xsd:annotation>
+                       <xsd:documentation>As described in the auditctl(8) manpage, the syscall can be any syscall name or number. The word 'all' may also be used.</xsd:documentation>
+                     </xsd:annotation>
+                   </xsd:element>
+				   <xsd:element name="permission" type="oval-def:EntityObjectAuditPermissionType" minOccurs="0" maxOccurs="1">
+                     <xsd:annotation>
+                       <xsd:documentation>As described in the auditctl(8) manpage, the permission describes the action that triggers the audit record.</xsd:documentation>
+                     </xsd:annotation>
+                   </xsd:element>
```

Comment:
I agree with you regarding syscall and field, I'll make the modifications.
Regarding -w, -a,-d, either I don't get it or it is against current mood in OVAL specification. It is not recommended to use options that modify the state of the audited system, right?

Reply:
@yuumasato Any thoughts on this?

Reply:
> Regarding -w, -a,-d, either I don't get it or it is against current mood in OVAL specification. It is not recommended to use options that modify the state of the audited system, right?

@ebouillard Yes, the scanner should not be making modifications to the system, but checking whether the system has an audit rule that adds  or removes a rule is not making changes to the system itself.
What can be confusing is that audit stores its rules in `.rules` files as a series of "commands", exactly like it would be done with `auditctl`command. Like, add/append this rule, or add/delete a watch.

Usually the rules in `.rules` files are defined with `-a` which will append a rule to the list of rules, but it is possible to define rules  with `-A`, which will add the rule to beginning of the list; or `-d` and `-W`, which removes matching rules and watches.

1. I think it is unlikely that a well defined `.rules` files would contain rules that remove others rules or watches. The admin could just remove the line that initialy added it.
2. And although `/etc/audit/audit.rules` shows all the audit rules,`auditctl -l` shows a final processed list of rules. It doesn't list the rule  additions at the beginning, it just lists it as the first rule with `-a`; and the rules with `-d` and `-W` are simply omitted, along with the rules/watches that were deleted.

Based on the 2 points above I believe we can ignore the rules with `-A`, `-d` and `-W`.
We should still account for the rules with `-w` though.